modules/sov-ibc/src/rpc/helpers.rs

```diff
@@ -13,22 +9,24 @@ use crate::context::IbcContext;
 use crate::Ibc;
 
 impl<S: Spec> Ibc<S> {
-    /// Determines the query height to use for the given request. If the query
-    /// height is not provided, it queries the host for the current height.
-    pub(super) fn determine_query_height(
+    pub(super) fn handle_request<F, Response>(
         &self,
-        query_height: Option<Height>,
+        request_height: Option<Height>,
         working_set: &mut WorkingSet<S>,
-    ) -> RpcResult<Height> {
-        match query_height {
-            Some(height) => Ok(height),
+        method: F,
+    ) -> Response
+    where
+        F: FnOnce(&IbcContext<'_, S>) -> Response,
+    {
+        match request_height {
+            Some(h) => {
+                let mut archival_working_set = working_set.get_archival_at(h.revision_height());
+                let ibc_ctx = IbcContext::new(self, &mut archival_working_set);
+                method(&ibc_ctx)
+            }
             None => {
-                let ibc_ctx = IbcContext {
-                    ibc: self,
-                    working_set: Rc::new(RefCell::new(working_set)),
-                };
-
-                ibc_ctx.host_height().map_err(to_jsonrpsee_error)
+                let ibc_ctx = IbcContext::new(self, working_set);
+                method(&ibc_ctx)
```

Comment:
This `working_set` is for query. wanted to make sure that it is a committed `working_set` - not yet-to-be committed one.

Reply:
update: I read carefully the conversation between you and Soares. It seems for `None` case, it is expected to read from yet-to-be committed `working_set`.